app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :create_cart
```

Comment:
But that can also be done the first time he adds an item to the shopping cart. If we don't have any cart created in session then we create one at that time. Every controller inherits from the `ApplicationController` and that means this method will be executed for every action we do in the application. It probably has no serious implication in this particular application but I'd be cautious about this kind of things.